src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/helpers.js

```diff
@@ -55,30 +55,32 @@ export function onNavBackKeepUrlParams({ goPreviousPath, urlParams }) {
 /**
  * Creates a path with a `add` query param
  * @param {Object} props
- * @param {string} props.basePath e.g. `/path-summary`
+ * @param {string} props.path e.g. `/path-item/:index`
  * @param {string | number} props.index
- * @returns {string} e.g. `/path-summary/0?add=true`
+ * @returns {string} e.g. `/path-item/0?add=true`
  */
-export function createArrayBuilderItemAddPath({ basePath, index }) {
-  return `${basePath}/${index}?add=true`;
+export function createArrayBuilderItemAddPath({ path, index }) {
+  return `${path.replace(':index', index)}?add=true`;
```

Comment:
I kinda wish we could use React router to pass state around instead of adding search parameters or needing to use local/session storage.

Reply:
hm, yeah that might be more clean and maintainable. If we have the need for it, we can refactor.